packages/next/build/compiler.ts

```diff
@@ -1,16 +1,21 @@
 import webpack, { Stats } from 'webpack'
 
 export type CompilerResult = {
-  errors: Error[]
-  warnings: Error[]
+  errors: string[]
+  warnings: string[]
 }
 
 function generateStats(result: CompilerResult, stat: Stats): CompilerResult {
-  const { errors, warnings } = stat.toJson({
-    all: false,
-    warnings: true,
-    errors: true,
-  })
+  const statsConfig = stat.compilation.compiler.options.stats
+  const statsConfigOrDefault = statsConfig
```

Comment:
What about allowing a partial override with something like this:
```typescript
const statsConfigOrDefault = Object.assign({
   all: false,
  warnings: true,
  errors: true,
}, statsConfig || {})
```

Reply:
[`statsConfig` may be an object or a string](https://webpack.js.org/configuration/stats#stats), so it can't be reliably spread.

For what it's worth, I made a similar fix over at storybookjs/storybook#6401 a couple months ago which seems to be working out.

Reply:
That makes sense :+1: